back/app/services/participation_permissions_service.rb

```diff
@@ -329,11 +338,11 @@ def disliking_limit_reached?(phase, user)
     phase.reacting_dislike_limited? && user.reactions.down.where(reactable: phase.ideas).size >= phase.reacting_dislike_limited_max
   end
 
-  def permission_denied_reason(user, action, phase)
-    permissions_service.denied_reason_for_resource user, action, phase
-  end
-
-  def permissions_service
-    @permissions_service ||= ::PermissionsService.new
+  def project_visible_disabled_reason(project, user)
```

Comment:
I think it would be nice to use this method in the ProjectPolicy in show.

Reply:
Going to open a separate ticket for this change - with so much change in this PR don't want to inadvertently break things with more unrelated code.